python/tests/test_microservice.py

```diff
@@ -18,6 +18,18 @@
 from google.protobuf import json_format
 
 
+def repeat_method(method, exception, args=(), kwargs={}, n=10, sleep=1):
```

Comment:
Just to keep it clean, would it make sense to use [`tenacity`](https://tenacity.readthedocs.io/en/latest/) instead of re-implementing the retry logic? We already use it on some of the integration tests:

https://github.com/SeldonIO/seldon-core/blob/7d1ecba612deb7d277e1064e0805460da374f070/testing/scripts/seldon_e2e_utils.py#L318-L331

Reply:
Probably true but then would have to factor out `stub.Predict` into a separate function and decorate it. I wanted to avoid doing too much restructuring but also provide better solution than just decorating the whole test as `flaky`.

Reply:
In general unit tests could use some refactoring to avoid code repetition - maybe it's good idea to use `tenacity` then?

Reply:
Actually, tenacity allows to retry [code block without the need to wrap it in an isolated function](https://tenacity.readthedocs.io/en/latest/#retrying-code-block) so I may give it a try now.